### PR TITLE
feat: Cache git repos and/or helm charts

### PR DESCRIPTION
**Pull request by arkaitzj**
While using helm-git in a manifest generation pipeline I have noticed a lot of redundant git fetches, helm indexes and helm packages.

With this PR, currently using it, 2 new behaviours become available.
* Cache git repos, by setting HELM_GIT_REPO_CACHE env to an existing folder it ensures repos are only fetched once per ref, it can also be used for offline generation by copying the cache as an artifact
* Cache helm charts, by setting HELM_GIT_CHART_CACHE env to an existing folder it ensures charts are only generated once, this speeds up manifest generating pipelines by a lot if git provided charts are reused in many places.

Even just `rm -rf caches && ./generate_manifests` is orders of magnitude faster for us when enabling both caches

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
       matrix:
         helm: ['2.17.0', '3.4.2', '3.7.1']
     env:
-      FIXTURES_GIT_BRANCH: ${{ github.head_ref || github.ref_name }}
+      FIXTURES_GIT_BRANCH: ${{ github.base_ref || github.ref_name }}
```
Comment (aslafy-z) [inline .github/workflows/release.yml:28]: Can you please revert this change, I just merged https://github.com/aslafy-z/helm-git/pull/223 that should solve the issue.

File: helm-git-plugin.sh [hunk 2]
```diff
@@ -49,6 +54,46 @@ git_try() {
   GIT_TERMINAL_PROMPT=0 git ls-remote "$_git_repo" --refs >&2 || return 1
 }
 
+
+
+#git_cache_intercept(git_repo, git_ref)
+git_cache_intercept(){
```
Comment (aslafy-z) [inline helm-git-plugin.sh:60]:
```suggestion
git_cache_intercept() {
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (arkaitzj) [reply .github/workflows/release.yml:28]: sure

**Review by aslafy-z — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -70,6 +70,8 @@ Pulling value files:
 --------|---------------|-----------
 `HELM_GIT_HELM_BIN`|Path to the `helm` binary. If not set, `$HELM_BIN` will be used|`helm`
 `HELM_GIT_DEBUG`|Increase `helm-git` log level to maximum|`0`
+`HELM_GIT_REPO_CACHE`|Path to use as a git repository cache to avoid fetching repos more than once or offline usage|`"" #cache disabled`
+`HELM_GIT_CHART_CACHE`|Path to use as a helm chart cache to avoid re-packaging or re-indexing charts|`"" #cache disabled`
```
Comment (aslafy-z) [inline README.md:74]:
```suggestion
`HELM_GIT_REPO_CACHE`|Path to use as a git repository cache to avoid fetching repos more than once or offline usage|`"" # disabled`
`HELM_GIT_CHART_CACHE`|Path to use as a helm chart cache to avoid re-packaging or re-indexing charts|`"" # disabled`
```

**Comment by aslafy-z** [comment]
Thank you @arkaitzj for this awesome addition! Looking forward to review your next contributions :100:

**Comment by arkaitzj** [comment]
my pleasure! thanks for speedy iterations! :)
